### PR TITLE
Fix psql_exec commands to work with different passwords.

### PR DESCRIPTION
**Pull request by telnoratti**
# RFI Checklist

### 1. Related Issues

- #3291 Install script fails if postgres master password and user password differ

---
### 2. Pull Request Checklist

- [ ]  Functionality completed as described in FRD
- [ ]  Dependencies, risks, assumptions in FRD addressed
- [ ]  Unit tests completed
- [ ]  Deployment requirements identified (e.g., SQL scripts, indexing)
- [ ]  Documentation completed
- [ ]  All code checkins completed

---
### 3. Review Checklist

_**After** the pull request has been submitted, fill out this section._

- [ ]  Code review completed or waived
- [ ]  Testing requirements completed
- [ ]  Usability testing completed or waived
- [ ]  Support testing completed or waived
- [ ]  Merged with develop branch and resolved conflicts

Connects to #3291.

### CODE REVIEW

File: scripts/installer/install [hunk 1]
```diff
@@ -829,8 +829,8 @@ else
     print "\nCreating Postgres database:\n";
     
     my $psql_command =
-	$psql_admin_exec
-	. "/createdb -h " . $CONFIG_DEFAULTS{'POSTGRES_SERVER'} . " -U postgres "
+	$psql_exec
+	. "/createdb -h " . $CONFIG_DEFAULTS{'POSTGRES_SERVER'} . " -U dvnapp "
```
Comment (pdurbin) [inline scripts/installer/install]: I'm just parroting something that @landreev said yesterday but we don't want "dvnapp" hard-coded here. It should come from `$CONFIG_DEFAULTS{'POSTGRES_USER'}`. See https://github.com/IQSS/dataverse/blob/v4.5/scripts/deploy/phoenix.dataverse.org/install#L8 for an example of where we use `export DB_USER=dvnapp` in a shell script to set this environment variable before calling the `install` script. We want to make sure this is configurable and not hard-coded.

**Comment by landreev** [comment]
@telnoratti - sorry to be slow, but why was the second change (lines 832-833) necessary? 

There was definitely a problem with the "create user" lines (that you have fixed, thank you!)
But I don't think I see any problem with issuing the "create database" command as the postgres admin user. I don't think we've ever had a problem with that part.

**Comment by telnoratti** [comment]
It's been a little while since I've dealt with this, but if I remember correctly without that change I got permission denied at that step. It may be a quirk of AWS's RDS. Let me double check that I'm remembering correctly and I'll provide error messages.

**Comment by telnoratti** [comment]
Here is an excerpt from the log of the install script without that change.

    Found Postgres psql command, version 9.2.15.
    (We'll be Using psql version 9.2)
    Checking if we can talk to Postgres as the admin user...
    Yes, we can!
    Configuring Postgres Database:
    Creating Postgres user (role) for the DVN:
    done.
    Creating Postgres database:
    Could not create Postgres database for the Dataverse app!
    (command: PGPASSWORD=root12345678; export PGPASSWORD; /usr/bin/createdb -h dpcb6ejtjaccjx.c1ovjmk8lyqm.us-east-1.rds.amazonaws.com -U postgres dvndb --owner=dvnapp)
    (psql exit code: 256)
    (STDOUT and STDERR: createdb: database creation failed: ERROR:  must be member of role "dvnapp"
    )
    aborting the installation (sorry!)

On AWS RDS the master user is not a true superuser. The install script does set all the necessary permissions for dvnapp to create the database. I'm not sure how other hosted postgresql instances would work outside of AWS, but this change should make it operate correctly as long as there is a postgres user available with the appropriate permissions.

**Comment by landreev** [comment]
@telnoratti ok, thanks - that explains it. I wasn't aware of this at all (of the "master user [not being] a true superuser". Good to know, and thanks for the report.

**Pull request merged**